pyro/nn/auto_reg_nn.py

```diff
@@ -140,15 +143,17 @@ class AutoRegressiveNN(nn.Module):
     def __init__(
             self,
             input_dim,
+            observed_dim,
             hidden_dims,
             param_dims=[1, 1],
             permutation=None,
             skip_connections=False,
             nonlinearity=nn.ReLU()):
-        super(AutoRegressiveNN, self).__init__()
+        super(ConditionalAutoRegressiveNN, self).__init__()
         if input_dim == 1:
-            warnings.warn('AutoRegressiveNN input_dim = 1. Consider using an affine transformation instead.')
+            warnings.warn('ConditionalAutoRegressiveNN input_dim = 1. Consider using an affine transformation instead.')
         self.input_dim = input_dim
+        self.observed_dim = observed_dim
```

Comment:
should we check that `observed_dim>0`?

Reply:
If I added `if observed_dim < 1: ...` into `ConditionalAutoRegressiveNN` then `AutoRegressiveNN` would throw the warning/error. There isn't a neat way to test the type in the initializer. The best I can think of is: `if type(self) == "<class 'pyro.nn.auto_reg_nn.ConditionalAutoRegressiveNN'>" and observed_dim < 1: ...`, but it seems like it isn't worth the trouble for a fragile solution (e.g. if you changed the module it was in it would break!)